### DISCUSSION TITLE
当无法安装简悦插件时的解决方案[已解决]

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1696] [重要通知]
# 🎉 简悦已更新到 2.3.0 版，适配 Manifest V3 方案，已上架 Chrome 应用商店

# 📥 Chrome 应用商店地址

https://chromewebstore.google.com/detail/simpread-reader-view/ijllcpnolfcooahcekpamkbidhejabll

# 🔗 讨论地址

https://github.com/Kenshin/simpread/issues/7326

# 🔌 插件系统 2.0

插件系统是这两个版本变化最大的地方，因为 Manifest V3 完全禁止了远超加载代码，导致插件系统改为「内置」方案。

2.2.0 → 在插件中心查看/安装插件
2.3.0 → 在简悦内启用/禁用插件（没有安装/卸载的功能）

关于插件系统 2.0 内置插件的更多介绍 https://github.com/Kenshin/simpread/issues/7329

# 🟩 带来的好处

- 当你安装插件，进入阅读模式后没有在「插件触发器」找到对应插件的问题。
- 更换为新设备时，插件配置无故「消失」的问题。
- 插件保存后但并没有生效的问题。
- 无法安装插件的问题。
- 导入配置文件后，但没有看到插件的问题。

也就是说 2.3.0 版彻底解决了插件各种问题。

***
***
***
***

## 重要通知

如果使用了 **同步助手的自动同步功能**，无法安装简悦插件的解决方案 https://github.com/Kenshin/simpread/discussions/2342

***

##  问题描述一

当进入 https://simpread.pro/plugins 并安装任意插件后，安装不上，安装异常或者无任何提示等情况。

### 解决方案

> 重置全部插件即可解决。

**选项页 → 插件管理 → 清除本地全部插件**

![image](https://user-images.githubusercontent.com/81074/104168764-84094080-5439-11eb-89e4-c3ce1d68cb29.png)

## 问题描述二

安装任何插件都会提示，获取失败，请稍后重试

![image](https://user-images.githubusercontent.com/21217483/150556388-07dddb44-219a-4c3d-beb9-1ca68996c075.png)

### 解决方案

权限设置造成的，请使用 https://github.com/Kenshin/simpread/discussions/1567 提供的方式解决。

### 关联问题

https://github.com/Kenshin/simpread/issues/3333#issuecomment-1022247377

**Comment by animerqi** [comment]
![image](https://user-images.githubusercontent.com/84447025/141777033-803f5218-5d0c-4766-be1b-0eb5495d5643.png)
安装的插件页面是这样的，点了没有反应。

**Reply by Kenshin** [reply]
解决方案看这里 https://github.com/Kenshin/simpread/discussions/2990

**Comment by zhongwencool** [comment]
<img width="877" alt="image" src="https://user-images.githubusercontent.com/3116225/158101849-38c0b405-19fd-490d-bfb5-8c009006e999.png">

一直处于正在安装。。。。
microsoft Edge
<img width="842" alt="image" src="https://user-images.githubusercontent.com/3116225/158101937-2d16cd7d-3b1f-4d9d-a804-25421e31db1c.png">

**Reply by Kenshin** [reply]
清除本地插件了吗？

**Reply by zhongwencool** [reply]
本地没有插件：
<img width="1453" alt="image" src="https://user-images.githubusercontent.com/3116225/158145347-a82bf6e8-10b9-4ad1-a180-fd848c640fbe.png">

**Reply by Kenshin** [reply]
嗨，这个界面没有显示插件，不代表本地 100% 没有插件，所以才能让先做清除本地插件的操作。
看现象似乎是你的浏览器阻止了接下来的「流程」。能否更换个浏览器试试？

如果还是不行的话，请把你的配置文件发送到 `help#####simpread.pro` 邮箱里面（注意 `#` → `@`）并注明这个 Issues ID：1696

**Reply by zhongwencool** [reply]
您好，清除本地全部插件的按键也试了，没用。
我已经于上周把配置发到你的help邮件了，麻烦帮我定位一下。

1. 换了chrome浏览器也是一样的错，
2. 然后重装了chrome（删除所有与chrome相关的cache），重装简悦扩展。还是一样的错。
3. 关掉所有网络代理。也还是一样。
<img width="1748" alt="image" src="https://user-images.githubusercontent.com/3116225/159154556-6f4fe426-510a-4c55-a07e-19689abde501.png">
<img width="1158" alt="image" src="https://user-images.githubusercontent.com/3116225/159154596-8dece950-88fd-4584-a32d-5d802db6db24.png">

就差没重装电脑了。
chrome版本：
```
Chrome 已是最新版本
版本 99.0.4844.74（正式版本） (x86_64)
```

**Reply by zhongwencool** [reply]
安装所有的插件都卡在这里，我还试了，连手机5G网络，还是不行。

**Reply by Kenshin** [reply]
嗨，不好意思，你的邮件被误判为垃圾邮件了。

我刚刚做了测试，使用你的配置文件导入是没问题的...
能否更换个浏览器（我是说非 Chrome ，试试 Edge）再试试？另外你访问 https://simpread.pro/plugins 界面的图标是否正常显示？

插件系统每天至少数千次的下载和安装，你应该是第一个有这种问题的。

**Reply by zhongwencool** [reply]
我报问题的时候就是Edge出来的问题。我看不行，
我才试的chrome。我的主力是Edge....
以前是装插件是正常的，我装成功过知乎的那个插件。我也不知道发生了什么....
访问插件页面是正常显示的。
<img width="1699" alt="image" src="https://user-images.githubusercontent.com/3116225/159165561-71ee8dd1-af4e-4491-bcb3-56dac5a23183.png">

**Reply by zhongwencool** [reply]
现在chrome和edge都是一样的现象。

**Reply by zhongwencool** [reply]
我终于成功了：我在这下面找到了灵感：
https://github.com/Kenshin/simpread/issues/3333
我同时装了 adBlock和uBlacklist， 禁用了adBlock，重启一下chrome就好了！
但是随后我想复现一下：再次找开AdBlock，发现也是成功了！无法复现 :(
说明不是100%出错的。
非常感谢这很好的工具，值得折腾 👍

**Reply by Kenshin** [reply]
😂

竟然是这个问题… 问题是我也有 uBlock... 

还好搞定了。 😉

**Reply by NihilityT** [reply]
我遇到了同样的问题，调试发现是 tabs.onUpdated 事件中 tab.url 为 undefined，导致后续流程无法走通。

![Screenshot 2022-03-24 224706](https://user-images.githubusercontent.com/21163191/159945628-f65e86bc-c140-4485-89ca-20be9c82f53f.png)

**Reply by NihilityT** [reply]
似乎与其他扩展有关，我在另一台存在这个问题的电脑上把除了 SimpRead 的扩展插件都给禁用后就可以安装了，不过再恢复插件启用状态就不重现了。
这个问题我已经提了一个 PR，经测试可以解决这个问题：https://github.com/Kenshin/simpread/pull/3592

**Reply by Kenshin** [reply]
谢谢 PR。

 💪  👍
